rust/headless-client/src/ipc_service.rs

```diff
@@ -167,7 +174,19 @@ struct Handler {
 
 enum Event {
     Callback(InternalServerMsg),
+    CallbackChannelClosed,
     Ipc(ClientMsg),
+    IpcDisconnected,
+    IpcError(anyhow::Error),
+    Terminate,
+}
+
+// Open to better names
+#[must_use]
+enum HandlerOutcome {
```

Comment:
I think something with `Result` would be more conventional, like:

```rust
enum HandlerOk {
    ClientDisconnected,
    ServiceTerminating
}
```

And then use a `Result<HandlerOk, ...>`